pkg/controller/node/nodecontroller.go

```diff
@@ -123,7 +126,9 @@ func NewNodeController(
 		kubeClient:             kubeClient,
 		recorder:               recorder,
 		podEvictionTimeout:     podEvictionTimeout,
-		podEvictor:             podEvictor,
+		maximumGracePeriod:     5 * time.Minute,
+		podEvictor:             NewRateLimitedTimedQueue(podEvictionLimiter, false),
+		terminationEvictor:     NewRateLimitedTimedQueue(podEvictionLimiter, false),
```

Comment:
I think they are effectively the same pool - the goal seems to be to
prevent the node controller from slaughtering the master during
negotiations eviction, and delete and terminate are two parts of the same
process.  Up to you, they don't seem separate to me.

On Aug 24, 2015, at 10:47 AM, Marek Grabowski <notifications@github.com>
wrote:

In pkg/controller/node/nodecontroller.go
<https://github.com/kubernetes/kubernetes/pull/12962#discussion_r37752204>:

> @@ -123,7 +126,9 @@ func NewNodeController(
>  		kubeClient:             kubeClient,
>  		recorder:               recorder,
>  		podEvictionTimeout:     podEvictionTimeout,
> -		podEvictor:             podEvictor,
> +		maximumGracePeriod:     5 * time.Minute,
> +		podEvictor:             NewRateLimitedTimedQueue(podEvictionLimiter, false),
> +		terminationEvictor:     NewRateLimitedTimedQueue(podEvictionLimiter, false),

Do we really want to share the same RateLimiter between podeEvictor and
terminationEvictor? I'm not against it, but in general I don't like sharing
state if it gives nothing except coding convenience.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/12962/files#r37752204>.